src/backend/README.md

```diff
@@ -1,174 +1,135 @@
-iterate through the previous flag ledger, and write any keys that are still present
-in the new flag ledger. This is done asynchronously.
-
-## Account Transactions
-rippled offers a RPC called `account_tx`. This RPC returns all transactions that
-affect a given account, and allows users to page backwards or forwards in time.
-Generally, this is a modeled with a table like so:
-```
-CREATE TABLE account_tx (
-account blob,
-ledger_sequence int,
-transaction_index int,
-hash blob,
-PRIMARY KEY(account,ledger_sequence,transaction_index))
+### `diff`
 ```
+    CREATE TABLE clio.diff (
+	    seq bigint,  # Sequence of the ledger version
+	    key blob,    # Hash of changes in the ledger version
+	    PRIMARY KEY (seq, key)
+    ) WITH CLUSTERING ORDER BY (key ASC) ...
+ ```
+This keyspace stores the object index of all the changes in each ledger version.
 
-An example of looking up from this table going backwards in time is:
-```
-SELECT hash FROM account_tx WHERE account = ? 
-    AND ledger_sequence <= ? and transaction_index <= ? 
-    ORDER BY ledger_sequence DESC, transaction_index DESC;
+### `account_tx`
 ```
+    CREATE TABLE clio.account_tx (
+	    account blob,
+	    seq_idx frozen<tuple<bigint, bigint>>,  # Tuple of (ledger_index, transaction_index)
+	    hash blob,                              # Hash of the transaction
+	    PRIMARY KEY (account, seq_idx)
+    ) WITH CLUSTERING ORDER BY (seq_idx DESC) ...
+ ```
+This keyspace stores the list of transactions made by each account. 
+
+
+### `successor`
+```
+    CREATE TABLE clio.successor (
+	    key blob,    # Object index
+	    seq bigint,  # The sequnce that this ledger object's predecessor and successor was updated
+	    next blob,   # Index of the next object that existed in this sequence
+	    PRIMARY KEY (key, seq)
+    ) WITH CLUSTERING ORDER BY (seq ASC) ...
+  ```
+This keyspace is the important backbone of how histories of ledger objects are stored in Cassandra. The successor table stores the object index of all ledger objects that were validated on the XRP network along with the ledger sequence that the object was upated on. Due to the unique nature of the table with each key being ordered by the sequence, by tracing through the table with a specific sequence number, Clio can recreate a Linked List data structure that represents all the existing ledger object at that ledger sequence. The special value of `0x00...00` and `0xFF...FF` are used to label the head and tail of the Linked List in the successor table. The diagram below showcases how tracing through the same table but with different sequence parameter filtering can result in different Linked List data representing the corresponding past state of the ledger objects. A query like `SELECT * FROM successor WHERE seq <= n ORDER BY seq DESC LIMIT 1;` can effectively trace through the successor table and get the Linked List of a specific sequence `n`.
+
+![Successor Table Trace Diagram](https://i.imgur.com/flG9FEy.png)
```

Comment:
We may want to add this .png file to display in the .md file, instead of just linking it.